processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -1095,6 +1096,26 @@ func (c *WatchClient) addOrUpdateReplicaSet(replicaset *apps_v1.ReplicaSet) {
 		c.ReplicaSets[string(replicaset.UID)] = newReplicaSet
 	}
 	c.m.Unlock()
+
+	for _, pod := range c.Pods {
```

Comment:
If we can avoid looping over Pod's cache by waiting for the all of the syncs to happen first that would be great. Left a comment back at the issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/37056#issuecomment-2577510756

Reply:
It seems like the solution is much simpler than initially thought. I now changed the order of the informer initialisation to start the replicaSet informer before the pod informer and that also fixed the issue. So far I have verified it manually, but I will look into providing an e2e test, similar to the current ones but with a collector restart.